pkg/virtual/workspaces/registry/rest.go

```diff
@@ -258,8 +258,61 @@ func (s *REST) authorizeOrgForUser(ctx context.Context, orgClusterName logicalcl
 	return nil
 }
 
+func (s *REST) authorizeForUser(ctx context.Context, orgClusterName logicalcluster.Name, user user.Info, verb string, resourceName string) error {
+	// Root org access is implicit for every user. For non-root orgs, we need to check for
+	// verb permissions against the clusterworkspaces/workspace sub-resource.
+	if orgClusterName == tenancyv1alpha1.RootCluster || sets.NewString(user.GetGroups()...).Has(kuser.SystemPrivilegedGroup) {
+		return nil
+	}
+
+	// Maintain compatibility with the old permission model until we completel remove
+	// the ability to have personal workspaces in top-level organizations
+	if parent, _ := orgClusterName.Split(); parent == tenancyv1alpha1.RootCluster {
+		switch verb {
+		case "get", "watch", "list":
+			return s.deprecatedAuthorizeOrgForUser(ctx, orgClusterName, user, "access")
+		case "create":
+			return s.deprecatedAuthorizeOrgForUser(ctx, orgClusterName, user, "member")
+		case "delete":
+			if err := s.deprecatedAuthorizeOrgForUser(ctx, orgClusterName, user, "access"); err != nil {
+				return err
+			}
```

Comment:
what about `update` and `patch` verbs?

Reply:
For now we don't support `update` or `patch` workspaces through the `workspaces` virtual workspace.

Reply:
Fixed in commit c05889057dc7234922ebf7b016a036c20e0b39a3